daemon/cluster/executor/container/attachment.go

```diff
@@ -19,8 +20,8 @@ type networkAttacherController struct {
 	closed  chan struct{}
 }
 
-func newNetworkAttacherController(b executorpkg.Backend, task *api.Task) (*networkAttacherController, error) {
-	adapter, err := newContainerAdapter(b, task)
+func newNetworkAttacherController(b executorpkg.Backend, task *api.Task, secrets exec.SecretGetter) (*networkAttacherController, error) {
+	adapter, err := newContainerAdapter(b, task, secrets)
```

Comment:
I think it was odd that the attacher just reused the container adapter. Hope we don't break something.

Leave in place for now.